File: Java/Laki2/H1/CharacterComparator.java
/* TIEP5.1 Lausekielinen ohjelmointi II
Harjoitus 1, tehtävä 6
Eveliina Toivanen (eveliina.toivanen@tuni.fi)
*/

import java.util.Scanner;
public class CharacterComparator{
    public static void main(String[] args) {
        
        Scanner lukija = new Scanner(System.in);

        System.out.println("Hello! I compare two characters of a string."); 
        System.out.println("Please, enter string:"); 
        String merkkijono = lukija.nextLine();

        System.out.println("Please, enter the first position:"); 
        int ekaIndeksi = lukija.nextInt();
        System.out.println("Please, enter the second position:");
        int tokaIndeksi = lukija.nextInt();
        
        int jonon_pituus = merkkijono.length();
        
        
        if ((0 <= ekaIndeksi && ekaIndeksi <= (jonon_pituus - 1)) 
            && (0 <= tokaIndeksi && tokaIndeksi <= (jonon_pituus - 1))) {
            char ekaMerkki = merkkijono.charAt(ekaIndeksi);
            char tokaMerkki = merkkijono.charAt(tokaIndeksi);
            
            if (ekaMerkki == tokaMerkki){
                System.out.println("'" + ekaMerkki + "' is equal to '" + tokaMerkki + "'.");
            }
            else {
                System.out.println("'" + ekaMerkki + "' is different from '" + tokaMerkki + "'.");
            }
        }
        else{
            System.out.println("Error!");
        }
    }
}

File: Python/LAKI1/H4/character_comparator.py
# TIEP1.1 Lausekielinen ohjelmointi I
# Harjoitus 4, tehtävä 5
# Eveliina Toivanen (eveliina.toivanen@tuni.fi)

# Ohjelma kysyy käyttäjältä merkkijonoa ja merkkijonon kahta indeksiä, joilta tämä
# haluaa tarkastaa, ovatko ne samanlaisia vai erilaisia.

indeksiarvo = True

print('Hello! I compare two characters of a string.')
print('Please, enter string:')
merkkijono = str(input())

print('Please, enter the first position:')
indeksi_1 = int(input())
print('Please, enter the second position:')
indeksi_2 = int(input())

# Tarkistetaan ovatko annetut indeksit laillisella indeksiarvojen välillä.
if not -len(merkkijono) <= indeksi_1 <= len(merkkijono) - 1:
    indeksiarvo = False
if not -len(merkkijono) <= indeksi_2 <= len(merkkijono) - 1:
    indeksiarvo = False

# Jos indeksit ovat sallittu, tarkistetaan, onko niiden paikalla samat merkit vai ei.
if indeksiarvo is True:
    if merkkijono[indeksi_1] == merkkijono[indeksi_2]:
        print('"', merkkijono[indeksi_1], '" is equal to "', merkkijono[indeksi_2], '".', sep='')
    elif merkkijono[indeksi_1] != merkkijono[indeksi_2]:
        print('"', merkkijono[indeksi_1], '" is different from "', merkkijono[indeksi_2],'".',
              sep='')

else:
    print('Error!')

